### PR TITLE
feat: maestro adapter

### PR DESCRIPTION
**Pull request by Vardominator**
* Abstracted `Adapter` interface
* Added `MaestroAdapter`, which implements `Adapter` interface
* Added example with `MaestroAdapter`
* Various helpers
* Various cleanup and formatting

### CODE REVIEW

**Comment by sourabhxyz** [comment]
Readme needs to be updated to reflect for additional adapter.

File: test/adapter.test.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import { BlockFrostAPI } from "@blockfrost/blockfrost-js";
+import { BlockFrostAPI } from '@blockfrost/blockfrost-js';
 
 import {
   ADA,
   Asset,
   BlockfrostAdapter,
```
Comment (sourabhxyz) [inline test/adapter.test.ts]: This file would not correctly compile as `BlockfrostAdapter` is longer exported now from `../src`, this breaking change should be avoided.

File: test/adapter.test.ts [hunk 2]
```diff
@@ -17,13 +17,13 @@ function mustGetEnv(key: string): string {
...
 
 const MIN_TESTNET =
-  "e16c2dc8ae937e8d3790c7fd7168d7b994621ba14ca11415f39fed724d494e";
+  'e16c2dc8ae937e8d3790c7fd7168d7b994621ba14ca11415f39fed724d494e';
 const MIN_MAINNET =
-  "29d222ce763455e3d7a09a665ce554f00ac89d2e99a1a83d267170c64d494e";
+  '29d222ce763455e3d7a09a665ce554f00ac89d2e99a1a83d267170c64d494e';
 const MIN_ADA_POOL_V1_ID_TESTNET =
-  "3bb0079303c57812462dec9de8fb867cef8fd3768de7f12c77f6f0dd80381d0d";
+  '3bb0079303c57812462dec9de8fb867cef8fd3768de7f12c77f6f0dd80381d0d';
```
Comment (sourabhxyz) [inline test/adapter.test.ts]: These styling related changes are leading to CI failure of format checks (`pnpm run check-format`).

File: test/adapter.test.ts
Comment (sourabhxyz) [inline test/adapter.test.ts:1]: We can abstract over this test to include for Maestro adapter, this should give confidence in changes introduced by this PR.

File: test/adapter.test.ts (hunk 2 above)
Comment (Vardominator) [reply test/adapter.test.ts]: fixed

File: test/adapter.test.ts
Comment (Vardominator) [reply test/adapter.test.ts:1]: done

**Pull request merged**